encoding/cache.go

```diff
@@ -83,7 +82,7 @@ func typeFields(t reflect.Type) []field {
 	next := []field{{typ: t}}
 
 	// Count of queued names for current level and the next.
-	count := map[reflect.Type]int{}
+	var count map[reflect.Type]int
```

Comment:
I would leave it as it is, but if we want to change it anyway, it would be better to have the following.

```go
count := make(map[reflect.Type]int, 0)
```

The `var count map[reflect.Type]int` would make an uninitialized map, so assigning any value to it right away will end up in a panic. Also, if we change this line, the next line should be changed too.

Reply:
count is assigned in line 96, if already assigned in 86, its never used.